modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -472,7 +504,7 @@ private void upgradeMetaPage(boolean inlineObjSupported) throws IgniteCheckedExc
                 lastIdxUsed++;
 
                 if (c != 0)
-                    return c;
+                    return fixSort(c, col.sortType);
```

Comment:
it could be breaking change from compatibility perspective

Reply:
not, it did not breaks compatibility, cause decision about which way to choose (left or right) is taken on the upper level (in BPlusTree). By the way I added compatibility test to check such behaviour